backend/src/main/java/com/team28/sidedish/controller/OAuthController.java

```diff
@@ -0,0 +1,60 @@
+package com.team28.sidedish.controller;
+
+import com.team28.sidedish.controller.dto.OAuthLoginResponse;
+import com.team28.sidedish.controller.oauth.OAuthAccessTokenRequest;
+import com.team28.sidedish.controller.oauth.OAuthUrl;
+import com.team28.sidedish.exception.LoginFailedException;
+import com.team28.sidedish.service.OAuthService;
+import com.team28.sidedish.utils.EnvUtils;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.net.URI;
+import java.util.Objects;
+
+@Controller
+@RequiredArgsConstructor
+@Slf4j
+public class OAuthController {
+
+    private static final String GITHUB_OAUTH_LOGIN_URL = "https://github.com/login/oauth/authorize";
+    private static final String GITHUB_OAUTH_SCOPE = "read:user";
+    private static final String GITHUB_OAUTH_RESPONSE_TYPE = "code";
+
+    private final OAuthService oAuthService;
+
+    @GetMapping("/auth/github")
+    public ResponseEntity<Void> githubAuthorization() {
+        String url = OAuthUrl.builder()
+                .baseUrl(GITHUB_OAUTH_LOGIN_URL)
+                .client_id(EnvUtils.getClientId())
+                .redirect_uri(EnvUtils.getRedirectUri())
+                .scope(GITHUB_OAUTH_SCOPE)
+                .response_type(GITHUB_OAUTH_RESPONSE_TYPE)
+                .build();
+
+        return ResponseEntity.status(HttpStatus.SEE_OTHER)
+                .location(URI.create(url))
+                .build();
```

Comment:
빌더 패턴을 이용해서 OAuth url을 생성해주셨네요 👍
복잡할수 있는 생성방식이 간단하게 표현된것 같습니다.

Reply:
builder를 통해서 String url을 생성해 주는게 아닌, OAuthUrl 인스턴스를 생성해줬으면 좀 더 좋았을것 같습니다.
```suggestion
        GithubOAuth githubOAuth = GithubOAuth.builder()
                .client_id(EnvUtils.getClientId())
                .redirect_uri(EnvUtils.getRedirectUri())
                .scope(GITHUB_OAUTH_SCOPE)
                .response_type(GITHUB_OAUTH_RESPONSE_TYPE)
                .build();

        return ResponseEntity.status(HttpStatus.SEE_OTHER)
                .location(URI.create(githubOAuth.url()))
                .build();
```

Reply:
추가로 서비스 레이어에 구현해도 좋았을 기능으로 보이네요 :)

Reply:
저는 단순히 웹 계층에서의 리다이렉션 처리라고만 생각했는데, 조금 더 세분화 하여 생각하면 인증 URL 을 만드는 로직 자체가 비즈니스로직 이라고 생각하고 서비스로직에도 넣을 수 있을 것 같네요!!
감사합니다!!